File: src/main/java/br/com/jiratorio/service/impl/FieldServiceImpl.java
package br.com.jiratorio.service.impl;

import br.com.jiratorio.client.FieldClient;
import br.com.jiratorio.domain.vo.JiraField;
import br.com.jiratorio.service.FieldService;
import java.util.List;
import lombok.extern.slf4j.Slf4j;
import org.springframework.cache.annotation.Cacheable;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Slf4j
@Service
public class FieldServiceImpl extends AbstractService implements FieldService {

    private final FieldClient fieldClient;

    public FieldServiceImpl(final FieldClient fieldClient) {
        this.fieldClient = fieldClient;
    }

    @Override
    @Cacheable("findAllFields")
    @Transactional(readOnly = true)
    public List<JiraField> findAllJiraFields() {
        log.info("Method=findAllJiraFields");
        return fieldClient.findAll(currentToken());
    }

}

File: src/main/java/br/com/jiratorio/service/impl/FieldServiceImpl.kt
package br.com.jiratorio.service.impl

import br.com.jiratorio.client.FieldClient
import br.com.jiratorio.domain.jira.JiraField
import br.com.jiratorio.extension.logger
import br.com.jiratorio.service.FieldService
import org.springframework.cache.annotation.Cacheable
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class FieldServiceImpl(
    private val fieldClient: FieldClient
) : FieldService {

    private val log = logger()

    @Cacheable("findAllFields")
    @Transactional(readOnly = true)
    override fun findAllJiraFields(): List<JiraField> {
        log.info("Method=findAllJiraFields")
        return fieldClient.findAll()
    }

}
